test/e2e/app-dir/app/next.config.js

```diff
@@ -8,6 +8,14 @@ module.exports = {
   // output: 'standalone',
   rewrites: async () => {
     return {
+      beforeFiles: [
+        {
+          source: '/before-files-rewrite-with-empty-arrays',
+          destination: '/',
+          has: [],
```

Comment:
`beforeFiles` with an empty `has` array is all that's needed to make the tests fail without the fix. I added the `missing` array, and did the same for `afterFiles` and `fallback` for completeness.